packages/components/layouts/tests/layouts.test.tsx

```diff
@@ -319,27 +297,46 @@ describe("<ZStack />", () => {
   })
 
   test("startIndex property works correctly", () => {
-    const { getByTestId } = render(
-      <ZStack data-testid="z-stack" startIndex={10}>
+    render(
+      <ZStack startIndex={10}>
         <Box>ZStack Item</Box>
       </ZStack>,
     )
 
-    const zStackItem = getByTestId("z-stack").firstChild
-    expect(zStackItem).toHaveStyle({ zIndex: 10 })
+    expect(screen.getByText("ZStack Item")).toHaveStyle({ zIndex: 10 })
   })
 
   test("Child elements are correctly overlaid and rendered", () => {
-    const { getByTestId } = render(
-      <ZStack data-testid="z-stack">
-        <Box data-testid="item-1">Item 1</Box>
-        <Box data-testid="item-2">Item 2</Box>
+    render(
+      <ZStack>
+        <Box>Item 1</Box>
+        <Box>Item 2</Box>
+      </ZStack>,
+    )
+
+    expect(screen.getByText("Item 1")).toHaveStyle({ position: "absolute" })
+    expect(screen.getByText("Item 2")).toHaveStyle({ position: "absolute" })
+  })
+
+  test("Whether the direction is working properly", () => {
+    render(
+      <ZStack direction="right">
+        <Box>Item 1</Box>
+        <Box>Item 2</Box>
       </ZStack>,
     )
+    expect(screen.getByText("Item 1")).toBeInTheDocument()
+    expect(screen.getByText("Item 2")).toBeInTheDocument()
+  })
 
-    const item1 = getByTestId("item-1")
-    const item2 = getByTestId("item-2")
-    expect(item1).toHaveStyle({ position: "absolute" })
-    expect(item2).toHaveStyle({ position: "absolute" })
+  test("Whether the reverse is working properly", () => {
+    render(
+      <ZStack reverse>
+        <Box>Item 1</Box>
+        <Box>Item 2</Box>
+      </ZStack>,
+    )
+    expect(screen.getByText("Item 1")).toBeInTheDocument()
+    expect(screen.getByText("Item 2")).toBeInTheDocument()
```

Comment:
I think we should add tests using toHaveStyle to confirm that the correct styles are being applied.

Reply:
I fixed [b839e02](https://github.com/yamada-ui/yamada-ui/pull/2257/commits/b839e02db39c7af8ff7c5f5dc210e0916af714af)